North-Seattle-College/ad440-winter2021-thursday-repo
# /users/{userId}/tasks Pagination

Closes #259 

Format of query string now accepted:
`/users/{userId}/tasks?count={count}&page={page}`

*note*: failing to pass in the count or page values will return 1st 10

**Other Changes:**
fix caching in /users

**To Test:**
- (1) in VSCode add the line `"CACHE_TOGGLE": "On"` to functions/local.settings.json
  - make sure this file also contains the variables ENV_DATABASE_CONNECTION_STRING, ENV_REDIS_KEY, ENV_REDIS_HOST, and ENV_REDIS_PORT (all available in key vault)
- (2) under the "Run/Debug" tab in VSCode, run the functions
- (3) go to Postman, c/p the endpoint for /users that VSCode will display on the terminal & enter integer values for {count} and {page}
- (4) run a GET 
  - you should see over {count} results in Postman
  - VSCode terminal should display the following:
  - ![image](https://user-images.githubusercontent.com/57076614/109789615-5c2a9380-7bc5-11eb-9ee8-13cd0ab64e35.png)
  - (if not, skip to step 6)
- (5) run another GET
  - VSCode terminal should display the following:
  -  ![image](https://user-images.githubusercontent.com/57076614/109786242-d22cfb80-7bc1-11eb-8fc9-f438b6793c21.png)
- (6) run a POST with a body like so:
   ```
   {
   	"firstName": "First",
	 "lastName": "Last",
	"email": "email@email.com"
   }
   ``` 
   - VSCode terminal should display the following:
   - ![image](https://user-images.githubusercontent.com/57076614/109786543-1cae7800-7bc2-11eb-9d37-1bad0cdc4ab7.png)
- (7) run (yet) another GET
   - VSCode terminal should display message from (4) again

- (8) Change the CACHE_TOGGLE value in /functions/local.settings.json to "Off"
- (9) run a GET
   - VSCode terminal should display the following:
   - ![image](https://user-images.githubusercontent.com/57076614/109787074-b2e29e00-7bc2-11eb-82c3-2413ae0637d0.png)
- (10) run a POST with a body like in (6)
   - VSCode terminal should display the following:
   - ![image](https://user-images.githubusercontent.com/57076614/109787355-fd641a80-7bc2-11eb-924c-14a9b49ac2eb.png)
 - (11) run a last GET
    - VSCode terminal should display the same as in (9)
